contracts/ExitFormat.sol

```diff
@@ -63,74 +72,30 @@ library ExitFormat {
-     * @param exitRequest An array with an entry for each asset: each entry is itself an array containing the exitRequest of the destinations to transfer funds to. Should be in increasing order. An empty array indicates "all".
+     * @notice Executes an exit by paying out assets and calling external contracts
+     * @dev Executes an exit by paying out assets and calling external contracts
+     * @param exit The exit to be paid out.
      */
-    function transfer(
-        ExitFormat.SingleAssetExit[] memory initialOutcome,
-        uint256[] memory initialHoldings,
-        uint48[][] memory exitRequest
-    )
-        internal
-        pure
-        returns (
-            ExitFormat.SingleAssetExit[] memory updatedOutcome,
-            uint256[] memory updatedHoldings,
-            ExitFormat.SingleAssetExit[] memory exit
-        )
-    {
-        require(initialOutcome.length == initialHoldings.length);
-        exit = new ExitFormat.SingleAssetExit[](initialOutcome.length);
-        for (uint256 i = 0; i < initialOutcome.length; i++) {
-            ExitFormat.Allocation[] memory initialAllocations =
-                initialOutcome[i].allocations;
-
-            updatedOutcome = new ExitFormat.SingleAssetExit[](
-                initialOutcome.length
-            );
-            updatedHoldings = initialHoldings;
-
-            uint48 k = 0;
-            uint256 surplus = initialHoldings[i];
-            ExitFormat.Allocation[] memory exitAllocations =
-                new ExitFormat.Allocation[](exitRequest[i].length);
-            for (uint256 j = 0; j < initialAllocations.length; j++) {
-                uint256 affordsForDestination =
-                    min(initialAllocations[j].amount, surplus);
-
-                if (
-                    exitRequest[i].length == 0 ||
-                    (k < exitRequest[i].length && exitRequest[i][k] == j)
-                ) {
-                    updatedHoldings[i] -= affordsForDestination;
-                    initialAllocations[j].amount -= affordsForDestination;
-                    exitAllocations[k] = ExitFormat.Allocation(
-                        initialAllocations[j].destination,
-                        affordsForDestination,
-                        initialAllocations[j].data
-                    );
-                    ++k;
-                } else {}
-                surplus -= affordsForDestination;
+    function executeExit(ExitFormat.SingleAssetExit[] memory exit) public {
+        for (uint256 i = 0; i < exit.length; i++) {
+            address asset = exit[i].asset;
+            for (uint256 j = 0; j < exit[i].allocations.length; j++) {
+                address payable destination =
+                    exit[i].allocations[j].destination;
+                uint256 amount = exit[i].allocations[j].amount;
+                address callTo = exit[i].allocations[j].callTo;
+                bytes memory data = exit[i].allocations[j].data;
+                if (asset == address(0)) {
```

Comment:
Can the `WithdrawHelper` handle this logic?

Reply:
A `WithdrawHelper` (as I understand it) may be used to forward ETH or tokens to a third party contract (as well as performing other arbitrary logic). So as long as the withdraw helper is called with a `CALL` (not `DELEGATECALL`), the funds need to move first into the `WithdrawHelper`. 

To maintain compatibility with the current `WithdrawHelpers`, we can stick with this for now.